ec/src/models/bls12/g2.rs

```diff
@@ -24,11 +21,15 @@ pub type G2Projective<P> = Projective<<P as Bls12Parameters>::G2Parameters>;
 pub struct G2Prepared<P: Bls12Parameters> {
     // Stores the coefficients of the line evaluations as calculated in
     // https://eprint.iacr.org/2013/722.pdf
-    pub ell_coeffs: Vec<EllCoeff<Fp2<P::Fp2Config>>>,
+    pub ell_coeffs: Vec<EllCoeff<P>>,
     pub infinity: bool,
 }
 
-pub(crate) type EllCoeff<F> = (F, F, F);
+pub(crate) type EllCoeff<P> = (
+    Fp2<<P as Bls12Parameters>::Fp2Config>,
+    Fp2<<P as Bls12Parameters>::Fp2Config>,
+    Fp2<<P as Bls12Parameters>::Fp2Config>,
+);
```

Comment:
I think this is a small change that we may just skip---we now have to be selective.

The main issue here is probably the naming of this local type. `F2` could be confusing, and maybe the easiest way is just to not create this local type since it is a one-time use.